packages/teleterm/src/ui/services/usageEvent/usageEventService.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ClusterOrResourceUri, ClusterUri, routing } from 'teleterm/ui/uri';
+import {
+  Cluster,
+  ReportEventRequest,
+  TshClient,
+} from 'teleterm/services/tshd/types';
+import { RuntimeSettings } from 'teleterm/mainProcess/types';
+import { ConfigService } from 'teleterm/services/config';
+import Logger from 'teleterm/logger';
+
+const logger = new Logger('UsageEventService');
+
+export class UsageEventService {
+  constructor(
+    private tshClient: TshClient,
+    private configService: ConfigService,
+    private findCluster: (clusterUri: ClusterUri) => Cluster,
+    private runtimeSettings: RuntimeSettings
+  ) {}
+
+  captureLogin(uri: ClusterOrResourceUri): Promise<void> {
+    const clusterParams = this.getClusterProperties(uri);
+    if (!clusterParams) {
+      return;
+    }
+    return this.reportUsageEvent({
+      loginEvent: {
+        clusterProperties: clusterParams,
+        arch: this.runtimeSettings.arch,
+        os: this.runtimeSettings.platform,
+        osVersion: this.runtimeSettings.osVersion,
+        connectVersion: this.runtimeSettings.connectVersion,
+      },
+    });
+  }
+
+  captureProtocolRun(
+    uri: ClusterOrResourceUri,
+    protocol: 'ssh' | 'kube' | 'db'
```

Comment:
In the RFD I was planning `proxy_db`, but I think we can just send `db`, we will match other teleport events.